riscv/mmu.cc

```diff
@@ -413,16 +415,17 @@ reg_t mmu_t::s2xlate(reg_t gva, reg_t gpa, access_type type, access_type trap_ty
         break;
       } else {
         reg_t ad = PTE_A | ((type == STORE) * PTE_D);
-#ifdef RISCV_ENABLE_DIRTY
-        // set accessed and possibly dirty bits.
+
         if ((pte & ad) != ad) {
-          pte_store(pte_paddr, pte | ad, gva, virt, type, vm.ptesize);
+          if (dirty_enabled) {
```

Comment:
Rather than adding the new `dirty_enabled` member, I think you could just check `proc->cfg->dirty_enabled` here.  (At this point in the code, we know that `proc` is non-null.)